src/pages/evm/wallet/SendPage.vue

```diff
@@ -1,21 +1,577 @@
+                return amount > 0 && amount <= +this.token.fullBalance;
+            } else {
+                return false;
+            }
+        },
+        isFormValid(): boolean {
+            return this.isAddressValid && this.isAmountValid && this.isPrecisionCorrect;
+        },
+        isPrecisionCorrect(): boolean {
+            if (this.token) {
+                if (this.useFiat) {
+                    const result = (this.amount.split('.')[1]?.length ?? 0) <= 2;
+                    return result;
+                } else {
+                    const result = (this.amount.split('.')[1]?.length ?? 0) <= this.token.decimals;
+                    return result;
+                }
+            } else {
+                return true;
+            }
+        },
+    },
+    methods: {
+        async updateEstimatedGas() {
+            const chain_settings = chainStore.loggedEvmChain?.settings;
+            if (chain_settings)  {
+                if (this.token?.isSystem) {
+                    this.estimatedGas = await chain_settings.getEstimatedGas(26250);
+                } else {
+                    this.estimatedGas = await chain_settings.getEstimatedGas(55500);
+                }
+            } else {
+                this.estimatedGas = { system: '0', fiat: '0' };
+            }
+        },
+        toggleUseFiat() {
+            if (this.token) {
+                if (this.useFiat) {
+                    this.amount = (+this.amount / this.token.price).toString();
+                } else {
+                    this.amount = (+this.amount * this.token.price).toFixed(2);
+                }
+            }
+
+            this.useFiat = !this.useFiat;
+        },
+        setAllBalance() {
+            if (this.token) {
+                if (this.useFiat) {
+                    this.amount = (+this.token.balance * this.token.price).toFixed(2);
+                } else {
+                    this.amount = this.token.balance;
+                }
+            } else {
+                this.amount = '';
+            }
+        },
+        viewTokenContract() {
+            if (this.token) {
+                const explorerUrl = ant.stores.chain.loggedEvmChain?.settings.getExplorerUrl();
+                if (explorerUrl) {
+                    window.open(explorerUrl + '/address/' + this.token.address, '_blank');
+                    return;
+                } else {
+                    ant.config.notifyErrorHandler(this.$t('settings.no_explorer', { network: ant.stores.chain.currentChain?.settings.getNetwork() }));
+                }
+            }
+        },
+        goBack() {
+            this.$router.back();
+        },
+        startTransfer() {
+            const token = this.token;
+            const amount = this.finalTokenAmount;
+            const to = this.address;
+            if (this.isFormValid) {
+                ant.stores.balances.transferTokens(token as Token, to, amount).then((trx: TransactionResponse) => {
```

Comment:
just double checking, ensure that `amount` is the proper type. does it need to be a bignumber?